### PR TITLE
exposing password encrypt utility interface as plugin

### PR DESCRIPTION
**Pull request by robertopaez**
**Is this a bugfix, or an enhancement? Please describe.**
Exposing Password Encrypt Utility as plugin (PasswordUtilityEncrypter). 

**Describe the solution you've implemented**
In order to have custom encrypt utilities, a new PasswordUtilityEncrypter plugin type was added .
The PasswordUtilityController npw also search the list of plugins implemented (along with ServiceLoader classes) 

**Describe alternatives you've considered**

**Additional context**

### CODE REVIEW

**Comment by robertopaez** [comment]
@gschueler

I decided add a new interface for the PasswordUtilityEncrypterPlugin.
Mainly because if we use the same PasswordUtilityEncrypter interfaz, the CommandLineSetup will be broken.
the controller now will support both implementations. For the plugin based on PasswordUtilityEncrypterPlugin, now the Descriptions  can be used.

File: core/src/main/java/com/dtolabs/rundeck/core/encrypter/PasswordUtilityEncrypterPlugin.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.dtolabs.rundeck.core.encrypter;
+
+import java.util.Map;
+
+public interface PasswordUtilityEncrypterPlugin {
+
+    Map encrypt(Map config);
```
Comment (gschueler) [inline core/src/main/java/com/dtolabs/rundeck/core/encrypter/PasswordUtilityEncrypterPlugin.java]: this interface is a bit crude for a plugin type, I would suggest something more like this:

```suggestion
    EncryptorResponse encrypt(Map<String,String> config);
```

where the `EncryptorResponse` is an interface like:

~~~
interface EncryptorResponse{
    boolean isValid()
    String getError()
    Map<String,String> getOutputs()
}
~~~

Comment (robertopaez) [reply core/src/main/java/com/dtolabs/rundeck/core/encrypter/PasswordUtilityEncrypterPlugin.java]: added @gschueler

**Comment by robertopaez** [comment]
@gschueler I added the docs that you asked

**Pull request merged**